arkane/common.py

```diff
@@ -268,19 +269,19 @@ def load_yaml(self, path, species, pdep=False):
                       'SingleExponentialDown': SingleExponentialDown,
                       'Wilhoit': Wilhoit,
                       'NASA': NASA,
+                      'NASAPolynomial': NASAPolynomial,
                       }
         freq_data = None
         if 'imaginary_frequency' in data:
             freq_data = data['imaginary_frequency']
-            del data['imaginary_frequency']
+            data = {key: data[key] for key in data.iterkeys() if key != 'imaginary_frequency'}
+        data['species'] = Species().fromSMILES(data['smiles'])
```

Comment:
sometimes an adjlist or InChI are given, and sometimes non of them

Reply:
I think this is something I have been struggling with a little bit. It is hard because we need an rmgpy Species to create an ArkaneSpecies, but because we don't have a species attribute on ArkaneSpecies the species object does not get saved to the yaml file. How should this work ideally? How can we reconstruct an ArkaneSpecies from a yaml file if the file does not have smiles or InChI?

Reply:
Settled by allowing a species to be created from SMILES, InChI, or adjacency list